client/subcomponents/Registration/registerHelper.js

```diff
@@ -4,38 +4,36 @@ import { setWaiting, clearWaiting } from '../../slices/waiting'
 import { setUser } from '../../slices/user'
 import { showError } from '../../slices/error'
 
-export function registerUser(
+export async function registerUser(
   user,
   isAdmin,
   authUser,
   navigateTo,
   consume = requestor
 ) {
-  const newUser = {
-    firstName: user.firstName,
-    lastName: user.lastName,
-    gardenId: user.gardenId,
-    email: authUser.email,
-    auth0Id: authUser.sub,
-  }
-  const storeState = getState()
-  const { token } = storeState.user
 
-  dispatch(setWaiting())
+  try {
+    const newUser = {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      gardenId: user.gardenId,
+      email: authUser.email,
+      auth0Id: authUser.sub,
+    }
+    const storeState = await getState()
+    const { token } = storeState.user
+    await dispatch(setWaiting())
```

Comment:
setWaiting() isn't async so you don't need to have await in front of this line

Reply:
Thank you!